tests/schemes/test_thrift.py

```diff
@@ -1037,3 +1038,83 @@ def testException(request):
     )
 
     assert 1 == res.status_code
+
+
+##############################################################################
+# Calling registered functions directly should be equivalent to calling them
+# as if they weren't registered at all
+
+
+def make_request(**kwargs):
+    """Constructs fake Request objects.
+
+    The ``_headers`` kwarg may be used to set the header on the Request
+    object.
+    """
+    request = mock.Mock()
+    for k, v in kwargs.items():
+        if k == '_headers':
+            request.headers = v
+        else:
+            setattr(request.body, k, v)
+    return request
+
+
+def test_void_call_directly(server, ThriftTest):
+
+    @server.thrift.register(ThriftTest)
+    def testVoid(request):
+        pass
+
+    resp = testVoid(make_request())
+    assert resp is None
+
+
+def test_void_with_headers_call_directly(server, ThriftTest):
+
+    @server.thrift.register(ThriftTest)
+    def testVoid(request):
+        assert request.headers == {'req': 'header'}
+        return Response(headers={'resp': 'header'})
+
+    resp = testVoid(make_request(_headers={'req': 'header'}))
```

Comment:
Tests are documentation. Any reason you can't just instantiate a Request instead of using mock?